.github/workflows/master-integration-tests.yml

```diff
@@ -0,0 +1,37 @@
+---
```

Comment:
Would it make sense to add this as a job in ci-tests.yaml? It seems to replicate most of the steps.
Can these tests used as a gate for the other automated tests? For instances, if this test fails, don't run the api testsuite (tyk-automated-tests). Not suggesting that we implement in this PR but just to get an idea of where this fits into the existing tests.

Reply:
This is an attempt to prefix the branch into the workflow file, to aid:

1. keeping all workflows on master (release-4-lts has it's own sync workflows which you don't find about if you look at master),
2. having structure in adding new integration tests (e.g. postgresql, etc.)

CI tests shares several responsibilities which could be managed with workflow_run, and I welcome concrete suggestions about cancelling or ordering the CI workflows and jobs. E.g. concurrency.cancel-in-progress is something we could add to all our CI jobs; https://sourcegraph.com/github.com/apache/shardingsphere/-/blob/.github/workflows/e2e-agent.yml?L53-55

Reply:
1. I think it is unfortunately necessary to have this workflow, and many others on release branches.
2. Doesn't ci-tests.yml provide the necessary structure?
Again, not saying that this needs to be solved in this PR but I see this workflow has to be kept in sync across release branches and sync-automation.yml is one way of solving it.